src/content/tutorials/github-twilio-notifications/index.md

```diff
@@ -0,0 +1,287 @@
+---
+async function githubWebhookHandler(request) {
+  if (request.method !== "POST") {
+    return simpleResponse(
+      200,
+      `Please send a POST request :)`
+    )
+  }
+  try{
+    if (!checkSignature(request)) {
+      return simpleResponse(403, "Incorrect Secret Code")
+    }
+
+    const formData = await request.json()
+    const repo_name = formData.repository.full_name
+    const action = await request.headers.get("X-GitHub-Event")
+    const sender_name = formData.sender.login
+  }
+  catch (e) {
+    return simpleResponse(
+      200,
+      `Error:  ${e} `
+    )
+  }
+}
+```
+
+The checkSignature function will use the crypto library to hash the received payload with our known secret key to ensure it matches the request hash. GitHub uses a HMAC hexdigest to compute the hash in the sha1 format [Github's Documentation](https://developer.github.com/webhooks/securing/).
+
+```javascript
+---
+filename: index.js
+---
+
+const crypto = require('crypto')
+
+async function createHexSignature(requestBody) {
+  let hmac = crypto.createHmac('sha1', SECRET_TOKEN)
+  hmac.update(requestBody,'utf-8')
+
+  return hmac.digest('hex')
+}
+
+async function checkSignature(request) {
+  let expectedSignature = await createHexSignature(await request.text())
+  let actualSignature = await request.headers.get("X-Hub-Signature")
+
+  return expectedSignature === actualSignature
+}
+```
+
+Since our project relies on importing a library, we need to also update our `wrangler.toml` and set `type = "webpack"`.
+
+
+## 3. Sending a Text with Twilio
+
+Finally, we will send a text message to you with the Github repo update info using Twilio. You'll need a Twilio account and number to send texts so follow [this guide to get setup](https://www.twilio.com/sms/api). FYI, Twilio has this [pretty awesome game](https://www.twilio.com/quest) where you can learn how to use their platform and get some free credits while you are at it.
+
+Then, we can create a helper function to send text messages for us by sending a post request to the Twilio API endpoint. [Refer to the Twilio reference here](https://www.twilio.com/docs/sms/api/message-resource#create-a-message-resource)
+
+We construct our headers and body in the format show n in the Twilio reference. Change the from value to your Twilio Phone number.
+
+```javascript
+---
+filename: index.js - sendText()
+---
+async function sendText(message){
+  let headers = new Headers();
+  headers.append("Authorization","Basic " + Buffer.from(accountSid + ":" + authToken).toString('base64'))
+  headers.append("Content-Type", "application/x-www-form-urlencoded")
+
+  const endpoint = "https://api.twilio.com/2010-04-01/Accounts/" + accountSid + "/Messages.json"
+
+  let encoded = new URLSearchParams()
+  encoded.append("To",recipient)
+  encoded.append("From", '+19388887573')
+  encoded.append("Body", message)
+
+  const request = {
+    body: encoded,
+    method: 'POST',
+    headers: headers
```

Comment:
generally you can simplify this by passing in a hash of key value pairs, instead of constructing headers at the beginning of the function:

```js
headers: {
  "Authorization": `Basic ${token}`,
  "Content-Type": "application/x-www-form-urlencoded"
}
```